### PR TITLE
Add new hook actionModuleUninstall before and after module uninstall

### PR DESCRIPTION
**Pull request by Nermendis**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add new hook actionModuleUninstall before and after module uninstall
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8868)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Hi @Nermendis,

thanks for your contribution!

I'm not opposed to the addition of new hooks. What is your need? What problems can be solved with this new hooks? Also, they need to be registered both in `hooks.xml` and `1.7.4.sql` files.

You can take a look at this [commit](https://github.com/PrestaShop/PrestaShop/pull/8368/commits/de53956a7e69795a84b365272a0a42af23040ee0) for an example.

Mickaël

File: classes/module/Module.php [hunk 1]
```diff
@@ -655,6 +655,7 @@ public static function getUpgradeStatus($module_name)
      */
     public function uninstall()
     {
+        Hook::exec('actionModuleUninstallBefore', array('object' => $this));
```
Comment (mickaelandrieu) [inline classes/module/Module.php:658]: new hooks need to be registered in hooks.xml and 1.7.4.0.sql file

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]
Need update of both xml and sql files

**Comment by Quetzacoalt91** [comment]
Hmm, I added Symfony events after these actions on modules, I wonder if we should add hooks as well.

It depends if the module can use the Symfony events. @mickaelandrieu?

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 not for now, as during the build of container Symfony may not be aware of classes autoloaded by your modules.

It will be available in 1.7.5 maybe.

**Comment by Quetzacoalt91** [comment]
Okay, let's do this like the hooks related to module installation.

@Nermendis can you please take some time to add the new hooks to the list as @mickaelandrieu requested?

**Comment by matks** [comment]
@Nermendis Sorry for late answer. It seems you have deleted your repository so I cannot retrieve the branch, but I have re-created a PR for you https://github.com/PrestaShop/PrestaShop/pull/18997

I will credit you

**Pull request closed without merging**